src/durableorchestrationstatus.ts

```diff
@@ -2,16 +2,61 @@ import { OrchestrationRuntimeStatus } from "./classes";
 import * as types from "durable-functions";
 
 export class DurableOrchestrationStatus implements types.DurableOrchestrationStatus {
+    public readonly name: string;
+    public readonly instanceId: string;
+    public readonly createdTime: Date;
+    public readonly lastUpdatedTime: Date;
+    public readonly input: unknown;
+    public readonly output: unknown;
+    public readonly runtimeStatus: OrchestrationRuntimeStatus;
+    public readonly customStatus?: unknown;
+    public readonly history?: Array<unknown>;
+
     /** @hidden */
-    constructor(
-        public readonly name: string,
-        public readonly instanceId: string,
-        public readonly createdTime: Date,
-        public readonly lastUpdatedTime: Date,
-        public readonly input: unknown,
-        public readonly output: unknown,
-        public readonly runtimeStatus: OrchestrationRuntimeStatus,
-        public readonly customStatus?: unknown,
-        public readonly history?: Array<unknown>
-    ) {}
+    constructor(init: unknown) {
+        if (!this.isDurableOrchestrationStatusInit(init)) {
+            throw new TypeError(
+                `Malformed data passed to DurableOrchestrationStatus constructor. Data received: ${JSON.stringify(
+                    init
+                )}`
+            );
+        }
+
+        this.name = init.name;
+        this.instanceId = init.instanceId;
+        this.createdTime = new Date(init.createdTime);
+        this.lastUpdatedTime = new Date(init.lastUpdatedTime);
+        this.input = init.input;
+        this.output = init.output;
+        this.runtimeStatus = init.runtimeStatus as OrchestrationRuntimeStatus;
+        this.customStatus = init.customStatus;
+        this.history = init.history;
+    }
+
+    private isDurableOrchestrationStatusInit(obj: unknown): obj is DurableOrchestrationStatusInit {
+        const objAsInit = obj as DurableOrchestrationStatusInit;
+        return (
+            objAsInit !== undefined &&
+            typeof objAsInit.name === "string" &&
+            typeof objAsInit.instanceId === "string" &&
+            (typeof objAsInit.createdTime === "string" || objAsInit.createdTime instanceof Date) &&
+            (typeof objAsInit.lastUpdatedTime === "string" ||
+                objAsInit.lastUpdatedTime instanceof Date) &&
+            objAsInit.input !== undefined &&
+            objAsInit.output !== undefined &&
+            typeof objAsInit.runtimeStatus === "string"
+        );
+    }
+}
+
+interface DurableOrchestrationStatusInit {

```

Comment:
nit: same feedback regarding `Init` suffix.

Reply:
You want it to be called `DurableOrchestrationStatusInitializer`? That feels unnecessarily long tbh. What else could `Init` mean? 

P.S. I was following `@azure/function`'s example here. E.g., see [`HttpResponseInit`](https://github.com/Azure/azure-functions-nodejs-library/blob/5e160396d9c431170e37b76ae06d1cbfdf2d1de5/types/http.d.ts#L197). Also `BodyInit` from `unidici`

Reply:
Well, if there's precedent for this in the `@azure/function` library, then I'm ok with it :)